modules/core/src/main/java/org/apache/ignite/internal/management/cache/CacheMetricsCommand.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.management.cache;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.internal.management.SystemViewCommand;
+import org.apache.ignite.internal.management.api.ComputeCommand;
+import org.apache.ignite.internal.visor.cache.metrics.VisorCacheMetricsTask;
+import org.apache.ignite.internal.visor.cache.metrics.VisorCacheMetricsTaskResult;
+import static java.util.Arrays.asList;
+import static org.apache.ignite.internal.visor.systemview.VisorSystemViewTask.SimpleType.STRING;
+
+/** */
+public class CacheMetricsCommand implements ComputeCommand<CacheMetricsCommandArg, VisorCacheMetricsTaskResult> {
+    /** {@inheritDoc} */
+    @Override public String description() {
+        return "Manages user cache metrics collection: enables, disables it or shows status";
+    }
+
+    /** {@inheritDoc} */
+    @Override public Class<CacheMetricsCommandArg> argClass() {
+        return CacheMetricsCommandArg.class;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Class<VisorCacheMetricsTask> taskClass() {
+        return VisorCacheMetricsTask.class;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void printResult(CacheMetricsCommandArg arg, VisorCacheMetricsTaskResult res, Consumer<String> printer) {
+        List<List<?>> values = new ArrayList<>();
+
+        try {
+            for (Map.Entry<String, Boolean> e : res.result().entrySet())
```

Comment:
Is there any reason for not using the stream api?

Reply:
Fixed.